Snowflake.Data/Core/SFStatement.cs

```diff
@@ -186,6 +186,12 @@ private SFBaseResultSet BuildResultSet(QueryExecResponse response, CancellationT
                 {
                     return new SFMultiStatementsResultSet(response.data, this, cancellationToken);
                 }
+
+                if (response.data.queryResultFormat == ResultFormat.Arrow)
+                {
+                    return new ArrowResultSet(response.data, this, cancellationToken);
+                }
+
                 return new SFResultSet(response.data, this, cancellationToken);

```

Comment:
Can we put comments this is JSON result? Also put it in "else" part to make it more obvious.